lib/os/cbprintf_nano.c

```diff
@@ -159,7 +159,7 @@ int cbvprintf(cbprintf_cb out, void *ctx, const char *fmt, va_list ap)
 				length_mod = 'H';
 			} else if (*fmt == 'l' && length_mod == 'l') {
 				length_mod = 'L';
-			} else if (length_mod == 0) {
+			} else if (length_mod == (char)0) {
```

Comment:
can you use `\0` like in `*fmt == '\0'` case?

Reply:
@nordic-auko I updated it. Now is `length_mod == '\0'`